src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/BlogPostForMailItem.java

```diff
@@ -1,29 +1,111 @@
 package pl.tomaszdziurko.jvm_bloggers.mailing;
 
-import lombok.Data;
+import com.google.common.base.Preconditions;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import lombok.Getter;
+import org.springframework.web.util.UriComponentsBuilder;
+import pl.tomaszdziurko.jvm_bloggers.UTMParametrsConstants;
 import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPost;
 import pl.tomaszdziurko.jvm_bloggers.blogs.domain.Blog;
 
-
-@Data
+/**
+ * {@link BlogPost} representation for email newsletter template purpose.
+ */
+@Getter
 class BlogPostForMailItem {
-
+    
+    private static final String UTM_SOURCE = "jvmbloggers";
+    private static final String UTM_MEDIUM = "newsletter";
+    
     private String title;
     private String url;
     private String authorLabel;
 
-    public BlogPostForMailItem(BlogPost blogPost) {
-        this.title = blogPost.getTitle();
-        this.url = blogPost.getUrl();
-        this.authorLabel = determineAuthorLabel(blogPost.getBlog());
+    /**
+     * Returns internal builder.
+     * 
+     * @return builder
+     */
+    public static Builder builder() {
+        return new Builder();
     }
+    
+    /**
+     * Builder for {@link BlogPostForMailItem}
+     */
+    public static class Builder {
```

Comment:
It would be nice to have methods like withDefaultUTMParameters/withUrlParameter available in builder. To my knowledge it is not possible to extend builder generated by lombok. https://github.com/peichhorn/lombok-pg has @Builder.Extension annotation. What do you say? Leave this as it is or use lombok-pg?